render/detailed/summary.go

```diff
@@ -93,52 +129,200 @@ func (n NodeSummary) Copy() NodeSummary {
-	)
-	if pid, ok := nmd.Latest.Lookup(process.PID); ok {
-		if !nameFound {
-			label = fmt.Sprintf("(%s)", pid)
+func processNodeSummary(base NodeSummary, n report.Node) (NodeSummary, bool) {
+	base.Label, _ = n.Latest.Lookup(process.Name)
+	base.Rank, _ = n.Latest.Lookup(process.Name)
+	base.Shape = Square
+
+	if p, ok := n.Counters.Lookup(report.Process); ok {
+		base.Stack = true
+		if p == 1 {
+			base.LabelMinor = fmt.Sprintf("%d process", p)
+		} else {
+			base.LabelMinor = fmt.Sprintf("%d processes", p)
+		}
+	} else {
+		pid, ok := n.Latest.Lookup(process.PID)
+		if !ok {
+			return NodeSummary{}, false
+		}
+		if containerName, ok := n.Latest.Lookup(docker.ContainerName); ok {
+			base.LabelMinor = fmt.Sprintf("%s (%s:%s)", report.ExtractHostID(n), containerName, pid)
+		} else {
+			base.LabelMinor = fmt.Sprintf("%s (%s)", report.ExtractHostID(n), pid)
 		}
-		id = render.MakeProcessID(report.ExtractHostID(nmd), pid)
 	}
-	_, isConnected := nmd.Latest.Lookup(render.IsConnected)
-	return baseNodeSummary(id, label, isConnected, nmd)
+
+	_, isConnected := n.Latest.Lookup(render.IsConnected)
+	base.Linkable = isConnected
+	return base, true
 }
 
-func containerNodeSummary(nmd report.Node) NodeSummary {
-	label, _ := render.GetRenderableContainerName(nmd)
-	containerID, _ := nmd.Latest.Lookup(docker.ContainerID)
-	return baseNodeSummary(render.MakeContainerID(containerID), label, true, nmd)
+func containerNodeSummary(base NodeSummary, n report.Node) (NodeSummary, bool) {
+	base.Label, _ = render.GetRenderableContainerName(n)
+
+	if c, ok := n.Counters.Lookup(report.Container); ok {
+		base.Stack = true
+		if c == 1 {
+			base.LabelMinor = fmt.Sprintf("%d container", c)
+		} else {
+			base.LabelMinor = fmt.Sprintf("%d containers", c)
+		}
+	} else {
+		base.LabelMinor = report.ExtractHostID(n)
+	}
+
+	if imageName, ok := n.Latest.Lookup(docker.ImageName); ok {
+		base.Rank = render.ImageNameWithoutVersion(imageName)
+	}
+
+	base.Shape = Hexagon
+	return base, true
 }
 
-func containerImageNodeSummary(nmd report.Node) NodeSummary {
-	imageName, _ := nmd.Latest.Lookup(docker.ImageName)
-	return baseNodeSummary(render.MakeContainerImageID(render.ImageNameWithoutVersion(imageName)), imageName, true, nmd)
+func containerImageNodeSummary(base NodeSummary, n report.Node) (NodeSummary, bool) {
+	imageName, ok := n.Latest.Lookup(docker.ImageName)
+	if !ok {
+		return NodeSummary{}, false
+	}
+
+	imageNameWithoutVersion := render.ImageNameWithoutVersion(imageName)
+	base.Label = imageNameWithoutVersion
+	base.Rank = imageNameWithoutVersion
+	base.Shape = Hexagon
+	base.Stack = true
+
+	if base.Label == "<none>" {
+		base.Label, _ = n.Latest.Lookup(docker.ImageID)
+		if len(base.Label) > 12 {
+			base.Label = base.Label[:12]
+		}
+	}
+
+	if i, ok := n.Counters.Lookup(report.ContainerImage); ok {
+		if i == 1 {
+			base.LabelMinor = fmt.Sprintf("%d image", i)
+		} else {
+			base.LabelMinor = fmt.Sprintf("%d images", i)
+		}
```

Comment:
When does this branch every get triggered?  When do we have groups with multiple container images?